KubeArmor/monitor/processTree.go

```diff
@@ -235,15 +236,15 @@ func (mon *SystemMonitor) GetParentExecPath(containerID string, ctx SyscallConte
 		if data, err := os.Readlink(filepath.Join(cfg.GlobalCfg.ProcFsMount, strconv.FormatUint(uint64(ctx.HostPPID), 10), "/exe")); err == nil && data != "" && data != "/" {
 			// // Store it in the ActiveHostPidMap so we don't need to read procfs again
 			// // We don't call BuildPidNode Here cause that will put this into a cyclic function call loop
-			// if pidMap, ok := ActiveHostPidMap[containerID]; ok {
-			// 	if node, ok := pidMap[ctx.HostPPID]; ok {
-			// 		node.ExecPath = data
-			// 		pidMap[ctx.HostPPID] = node
-			// 	} else if node, ok := pidMap[ctx.HostPID]; ok {
-			// 		node.ExecPath = data
-			// 		pidMap[ctx.HostPID] = node
-			// 	}
-			// }
+			if pidMap, ok := ActiveHostPidMap[containerID]; ok {
+				if node, ok := pidMap[ctx.HostPPID]; ok {
+					node.ExecPath = data
+					pidMap[ctx.HostPPID] = node
+				} else if node, ok := pidMap[ctx.HostPID]; ok {
+					node.ParentExecPath = data
```

Comment:
@Aryan-sharma11 can you help me understand this change?

Reply:
We are implementing caching for Parent Process. It will either be available on it's own PID (which is PPID) or ParentExecPath on the actual process PID.

Check the code above for reference

This is how it's checked

```go
if pidMap, ok := ActiveHostPidMap[containerID]; ok {
		if node, ok := pidMap[ctx.HostPID]; ok {
			path = node.ParentExecPath
			if path != "/" && strings.HasPrefix(path, "/") {
				return path
			}
		}
		// check if parent pid node exists
		if node, ok := pidMap[ctx.HostPPID]; ok {
			path = node.ExecPath
			if path != "/" && strings.HasPrefix(path, "/") {
				return path
			}
		}
	}
```